File: MSCL_Examples/Inertial/CSharp/MSCL_Inertial_Example_CSharp/MSCL_Inertial_Example_CSharp/getCurrentConfig.cs
using System;

namespace MSCL_Inertial_Example_CSharp
{
    class Example1
    {
        public static void getCurrentConfig(mscl.InertialNode node)
        {
            //many other settings are available than shown below
            //reference the documentation for the full list of commands

            //if the node supports AHRS/IMU
            if (node.features().supportsCategory(mscl.MipTypes.DataClass.CLASS_AHRS_IMU))
            {
                //get a list of the AHRS/IMU channels currently active on the Node
                mscl.MipChannels ahrsImuActiveChs = node.getActiveChannelFields(mscl.MipTypes.DataClass.CLASS_AHRS_IMU);

                Console.WriteLine("AHRS/IMU Channels");
                Console.WriteLine("-----------------");
                foreach (mscl.MipChannel ch in ahrsImuActiveChs)
                {
                    Console.WriteLine("Channel Field: " + ch.channelField().ToString("X"));
                    Console.WriteLine("Sample Rate: " + ch.sampleRate().prettyStr());
                }
            }

            //if the node supports Estimation Filter
            if (node.features().supportsCategory(mscl.MipTypes.DataClass.CLASS_ESTFILTER))
            {
                //get a list of the Estimation Filter channels currently active on the Node
                mscl.MipChannels estFilterActiveChs = node.getActiveChannelFields(mscl.MipTypes.DataClass.CLASS_ESTFILTER);

                Console.WriteLine();
                Console.WriteLine("Estimation Filter Channels");
                Console.WriteLine("--------------------------");
                foreach (mscl.MipChannel ch in estFilterActiveChs)
                {
                    Console.WriteLine("Channel Field: " + ch.channelField().ToString("X"));
                    Console.WriteLine("Sample Rate: " + ch.sampleRate().prettyStr());
                }
            }

            //if the node supports GNSS
            if (node.features().supportsCategory(mscl.MipTypes.DataClass.CLASS_GNSS))
            {
                //get a list of the GNSS channels currently active on the Node
                mscl.MipChannels gnssActiveChs = node.getActiveChannelFields(mscl.MipTypes.DataClass.CLASS_GNSS);

                Console.WriteLine();
                Console.WriteLine("GNSS Channels");
                Console.WriteLine("--------------------------");
                foreach (mscl.MipChannel ch in gnssActiveChs)
                {
                    Console.WriteLine("Channel Field: " + ch.channelField().ToString("X"));
                    Console.WriteLine("Sample Rate: " + ch.sampleRate().prettyStr());
                }
            }

            Console.WriteLine("Altitude Aiding enabled?: " + node.getAltitudeAid().ToString());

            mscl.PositionOffset offset = node.getAntennaOffset();
            Console.WriteLine("Antenna Offset: x=" + offset.x().ToString() + " y=" + offset.y().ToString() + " z=" + offset.z().ToString());

            Console.WriteLine("Pitch/Roll Aiding enabled?: " + node.getPitchRollAid().ToString());
        }
    }
}

File: MSCL_Examples/Inertial/Python/getCurrentConfig.py
# Updated: 2/21/2024

import sys

# TODO: change this path to match your setup
# import the mscl library.
arch = sys.maxsize > 2**32 and "x64" or "x32"
sys.path.append("../../../Output/Python{sys.version_info.major}.{sys.version_info.minor}/{arch}/Release")

import mscl

# TODO: change these constants to match your setup
COM_PORT = "COM4"

try:
    #create a Serial Connection with the specified COM Port, default baud rate of 921600
    connection = mscl.Connection.Serial(COM_PORT)

    #create an InertialNode with the connection
    node = mscl.InertialNode(connection)

    #many other settings are available than shown below
    #reference the documentation for the full list of commands

    #if the node supports AHRS/IMU
    if node.features().supportsCategory(mscl.MipTypes.CLASS_AHRS_IMU):
        #get a list of the AHRS/IMU channels currently active on the Node
        ahrsImuActiveChs = node.getActiveChannelFields(mscl.MipTypes.CLASS_AHRS_IMU)

        print("AHRS/IMU Channels")
        print("-----------------")
        for ch in ahrsImuActiveChs:
            print("Channel Field:", ch.channelField())
            print("Sample Rate:", ch.sampleRate().prettyStr())

    #if the node supports Estimation Filter
    if node.features().supportsCategory(mscl.MipTypes.CLASS_ESTFILTER):
        #get a list of the Estimation Filter channels currently active on the Node
        estFilterActiveChs = node.getActiveChannelFields(mscl.MipTypes.CLASS_ESTFILTER)

        print("Estimation Filter Channels")
        print("--------------------------")
        for ch in estFilterActiveChs:
            print("Channel Field:", ch.channelField())
            print("Sample Rate:", ch.sampleRate().prettyStr())

    #if the node supports GNSS
    if node.features().supportsCategory(mscl.MipTypes.CLASS_GNSS):
        #get a list of the GNSS channels currently active on the Node
        gnssActiveChs = node.getActiveChannelFields(mscl.MipTypes.CLASS_GNSS)

        print("GNSS Channels")
        print("--------------------------")
        for ch in gnssActiveChs:
            print("Channel Field:", ch.channelField())
            print("Sample Rate:", ch.sampleRate().prettyStr())

    print("Altitude Aiding enabled?:", node.getAltitudeAid())

    offset = node.getAntennaOffset()
    print("Antenna Offset: x=", offset.x(), " y=", offset.y(), " z=", offset.z())

    print("Pitch/Roll Aiding enabled?:", node.getPitchRollAid())

except mscl.Error as e:
    print("Error:", e)
